File: test/single-language/php/phpcli/http-service/success-json-api/src/Configuration.php
<?php
declare(strict_types=1);
namespace Fixture;
final class Configuration {
    public function __construct(public readonly string $mode, public readonly int $status, public readonly string $label) {}
    public static function load(): self {
        $mode = 'json';
        $label = getenv('FIXTURE_LABEL');
        return new self($mode, 200, $mode === 'config'
            ? ($label === false ? 'runtime-config-default' : $label) : 'deployment-smoke-ok');
    }
}

File: test/single-language/kotlin/gradle/http-service/success-json-api/src/main/kotlin/acceptance/Configuration.kt
package acceptance
data class Configuration(val port: Int, val mode: String, val status: Int, val label: String) {
    companion object {
        fun load(): Configuration {
            val raw = System.getenv("PORT") ?: ""
            val port = raw.toIntOrNull()
            require(raw.matches(Regex("[0-9]{1,5}")) && port != null && port in 1..65535) {
                "Invalid PORT"
            }
            val mode = "json"
            val label =
                if (mode == "config") System.getenv("FIXTURE_LABEL") ?: "runtime-config-default"
                else "deployment-smoke-ok"
            return Configuration(port, mode, 200, label)
        }
    }
}
